### PR TITLE
fix(router-link): add missing target prop

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: https://github.com/ionic-team/ionic/issues/18655

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/router-link/router-link.tsx [hunk 1]
```diff
@@ -36,6 +36,13 @@ export class RouterLink implements ComponentInterface {
    */
   @Prop() routerDirection: RouterDirection = 'forward';
 
+  /**
+   * Specifies where to display the linked URL.
+   * Only applies when an `href` is provided.
+   * Special keywords: `"_blank"`, `"_self"`, `"_parent"`, `"_top"`.
+   */
+  @Prop() target: string | undefined;
```
Comment (manucorporat) [inline core/src/components/router-link/router-link.tsx]: wondering why not using `?`?
```suggestion
  @Prop() target?: string;
```

File: core/src/components/datetime/datetime.tsx [hunk 2]
```diff
@@ -565,8 +565,7 @@ export class Datetime implements ComponentInterface {
   }
 
   private hasValue(): boolean {
-    const val = this.datetimeValue;
-    return Object.keys(val).length > 0;
```
Comment (manucorporat) [inline core/src/components/datetime/datetime.tsx]: is this related to the change?

File: core/src/components/router-link/router-link.tsx (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/router-link/router-link.tsx]: Because this is how we have it done for all other components using `AnchorInterface` and if we did decide to add an interface for this it would already be working. https://github.com/ionic-team/ionic/pull/17883/files#diff-36f66a20486cf32eda71ad087e5998bcR55

File: core/src/components/datetime/datetime.tsx (hunk 2 above)
Comment (brandyscarney) [reply core/src/components/datetime/datetime.tsx]: Mm no. I branched off of master and am comparing to the hotfix. Must've accidentally pulled in an unrelated fix, good catch!

File: core/src/components/router-link/router-link.tsx (hunk 1 above)
Comment (manucorporat) [reply core/src/components/router-link/router-link.tsx]: Gotcha!
